sinonjs/sinon
# Throw meaningful error stubbing ECMAScript Module

#### Purpose (TL;DR)
Throw a meaningful error when encountering ES Modules in `sinon.stub()`

#### Background
Ref #1711 and #1623 for background.

This feature is simply about throwing a meaningful error, instead of one that tells the user that the property cannot be deleted. As the exports from ECMAScript modules are immutable once created we
cannot do anything about them from the view of Sinon, so it's preferable to simply tell the user that when we can.

This should not affect transpiled code (say, using Babel), as the resulting modules are no longer true ECMAScript modules (but rather some form of CommonJS module).

#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm run test-es-module` (included in `npm test`)

#### Checklist for author
